### PR TITLE
Add native 2I queries to PBC protocol

### PR DESCRIPTION
**Pull request by seancribbs**
This is part of a series of pull-requests that adds native 2I queries to the Protocol Buffers interface.  In my unscientific tests, the native query shows a 50-80% improvement over the emulation via MapReduce. (See https://gist.github.com/f5d4599cff9e2f9f9c91)

Related: basho/riak_pb#1

**NOTE**: This requires the PBC refactoring (#321) to be present first.

### CODE REVIEW

File: src/riak_index.erl [hunk 1]
```diff
@@ -227,6 +228,40 @@ timestamp() ->
     {MegaSeconds,Seconds,MilliSeconds}=erlang:now(),
     (MegaSeconds * 1000000000000) + (Seconds * 1000000) + MilliSeconds.
 
+%% @spec to_index_op_query(binary(), [binary()]) ->
```
Comment (kellymclaughlin) [inline src/riak_index.erl]: I know this is not something you did directly, but the function name in the spec doesn't match the actual function name.

**Comment by kellymclaughlin** [comment]
Diff between this branch and `sdc-pbc-extract`: https://github.com/basho/riak_kv/compare/sdc-pbc-extract...sdc-pb-2i

File: src/riak_kv_pb_index.erl [hunk 2]
```diff
@@ -0,0 +1,103 @@
+%% -------------------------------------------------------------------
+%%
+%% riak_kv_pb_object: Expose KV functionality to Protocol Buffers
```
Comment (kellymclaughlin) [inline src/riak_kv_pb_index.erl]: s/riak_kv_pb_object/riak_kv_pb_index

**Comment by seancribbs** [comment]
@kellymclaughlin Fixed typos and also followed @rzezeski's suggestion about using `#state{}` instead of the crazy macro.

**Comment by kellymclaughlin** [comment]
HTTP and mapreduce 2I queries still work for me. The changes from `sdc-pbc-extract` look good. Just wondering if there is a straightforward way to exercise `riak_kv_pb_socket` that I have overlooked.

**Comment by kellymclaughlin** [comment]
+1 to merge

**Pull request merged**